src/pages/matrix.tsx

```diff
@@ -1,62 +1,246 @@
+  if (!matrixData || !colorScaleMap || !styles) {
     return <ChaiseSpinner />;
   }
 
-  return <Matrix {...matrixProps} />;
+  const { gridDataMap, gridData, legendData, options } = matrixData;
+
+  /**
+   * Handles changes to the search bar
+   *
+   * @param option
+   */
+  const handleChange = (option: any) => {
+    if (option) {
+      setInput(option.label);
+      scrollToInput(option.label);
+    } else {
+      setInput('');
+      gridRef.current.clearSearch();
+    }
+  };
+
+  /**
+   * Handles input changes to the search bar
+   *
+   * @param value
+   * @param action
+   */
+  const handleInputChange = (value: string, action: InputActionMeta) => {
+    if (action.action === 'input-change' || action.action === 'set-value') {
+      setInput(value);
+    }
+  };
+
+  /**
+   * Sets a state to show message when no results found from search
+   */
+  const showNoResults = () => {
+    setToastMessage('no result found');
+    setTimeout(() => setToastMessage(''), 2000);
+  };
+
+  /**
+   * Scrolls the grid to the given input string if found in the matrix,
+   * otherwise shows a no results message
+   *
+   * @param currInput
+   * @returns
+   */
+  const scrollToInput = (currInput: string | undefined) => {
+    if (!currInput) {
+      return;
+    }
+    if (!gridDataMap[currInput.toLowerCase()]) {
+      showNoResults();
+      return;
+    }
+
+    const selected = gridDataMap[currInput.toLowerCase()];
+    if (selected.type === 'row') {
+      gridRef.current.searchRow(selected.index);
+    } else if (selected.type === 'col') {
+      gridRef.current.searchCol(selected.index);
+    }
+  };
+
+  /**
+   * Handles when the user clicks the search button
+   */
+  const handleSubmit = () => {
+    scrollToInput(input);
+  };
+
+  /**
+   * Handls changes to the Color theme dropdown
+   *
+   * @param option
+   */
+  const handleChangeTheme = (option: any) => {
+    if (option) {
+      setColorThemeOption(option);
+    }
+  };
+
+  const numRows = gridData.length;
+  const numColumns = gridData[0].length;
+
+  // Calculate Layout of the Grid:
+
+  const maxCols = styles.maxCols;
+  const maxRows = styles.maxRows;
+
+  const rowHeaderWidth = styles.rowHeaderWidth;
+  const colHeaderHeight = styles.colHeaderHeight;
+  const cellHeight = styles.cellHeight;
+  const cellWidth = styles.cellWidth;
+
+  const widthBufferSpace = 50; // buffer space for keeping everything in viewport
+  const heightBufferSpace = 330; // buffer space for keeping everything in viewport
+  const gridHeight = Math.min(
+    cellHeight * numRows, // can't exceed total grid
+    height - colHeaderHeight - heightBufferSpace, // can't exceed browser height
+    colHeaderHeight + cellHeight * maxRows
+  );
+  const gridWidth = Math.min(
+    cellWidth * numColumns, // can't exceed total grid
+    width - rowHeaderWidth - widthBufferSpace, // can't exceed browser width
+    rowHeaderWidth + cellWidth * maxCols
+  );
+
+  const gridTitle = styles.title;
```

Comment:
As I mentioned we want to use `title_markdown` and `subtitle_markdown`. You should make sure you're handling where these two are missing/invalid.

And to parse markdown you would do the following:

```ts
const title = ConfigService.ERMrest.renderMarkdown(title_markdown);

 // add title only if it's defined
if (title && title.length > 0) {

}
```

Reply:
modified